waiter/src/waiter/scheduler.clj

```diff
@@ -73,6 +73,18 @@
 (defn make-ServiceInstance [value-map]
   (map->ServiceInstance (merge {:extra-ports [] :flags #{}} value-map)))
 
+(let [name-key :waiter/scheduler-name]
+  (defn attach-name
```

Comment:
Why is this a separate function, invoked separately from the scheduler creation? Would it be better if `utils/create-component` attached the meta-data instead?

Reply:
Do we want all components to get names? This means that usages of the `scheduler` now expect the scheduler to have been created with the `create-component` method and invoke a `component->name` function to get the name. Would/Should all schedulers be created as components?

My vote would be to leave the component naming restricted to scheduler instances and generify later for other components when we need it.